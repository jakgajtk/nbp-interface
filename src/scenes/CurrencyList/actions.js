const REFRESH_LIST = 'LIST/REFRESH'

export const actions = {
  REFRESH_LIST
}

export const refresh = data => ({
  type: REFRESH_LIST,
  data
})
