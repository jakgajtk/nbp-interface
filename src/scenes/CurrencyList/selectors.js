export default function getCurrencyList(state) {
  return state.currencyList.list
}