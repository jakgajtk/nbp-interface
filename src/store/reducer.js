import { reducer as bees } from 'redux-bees'
import { combineReducers } from 'redux'

import CurrencyDetailReducer from '../scenes/CurrencyDetail/reducer'
import CurrencyListReducer from '../scenes/CurrencyList/reducer'
import FavouriteReducer from '../scenes/Favourite/reducer'

export default combineReducers({
  bees,
  currencyDetail: CurrencyDetailReducer,
  currencyList: CurrencyListReducer,
  favourite: FavouriteReducer
})
