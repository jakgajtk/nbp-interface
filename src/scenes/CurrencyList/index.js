import React, { Component } from 'react'
import { connect } from 'react-redux'
import api from '../../services/Api'
import { refresh } from './actions'
import { List } from 'material-ui/List'
import getCurrencyList from './selectors'
import Subheader from 'material-ui/Subheader'
import RaisedButton from 'material-ui/RaisedButton'
import CurrencyListItem from '../../components/CurrencyListItem'

const PRIMARY_TABLE = 'A'
const SECONDARY_TABLE = 'B'
const style = {
  float: 'right',
  'lineHeight': '30px'
}

class CurrencyList extends Component {
  constructor (props) {
    super(props)
    this.state = {table: PRIMARY_TABLE}
  }

  componentDidMount () {
    if (!this.props.currencyList.length) {
      this.getPrimaryList()
    }
  }

  getPrimaryList = () => {
    const { dispatch } = this.props
    dispatch(api.getPrimaryCurrencyList())
      .then(data => {
        this.setState({table: PRIMARY_TABLE})
        dispatch(refresh(data.body[0].rates))
      })
      .catch(error => {
        console.log(error)
      })
  }

  getSecondaryList = () => {
    const { dispatch } = this.props
    dispatch(api.getSecondaryCurrencyList())
      .then(data => {
        this.setState({table: SECONDARY_TABLE})
        dispatch(refresh(data.body[0].rates))
      })
      .catch(error => {
        console.log('Error while fetching data: ', error)
      })
  }

  render () {
    const result = this.props.currencyList.map(item => <CurrencyListItem item={item} table={this.state.table} key={item.code} />)
    return (
      <div>
        <List>
          <Subheader>
            <span>Currency Rates List</span>
            <span style={style}>
              <RaisedButton label='Primary Currency Rates' primary onClick={this.getPrimaryList} />
              <RaisedButton label='Secondary Currency Rates' secondary onClick={this.getSecondaryList} />
            </span>
          </Subheader>
          {result}
        </List>
      </div>
    )
  }
}

export default connect(state => ({
  currencyList: getCurrencyList(state)
}))(CurrencyList)
