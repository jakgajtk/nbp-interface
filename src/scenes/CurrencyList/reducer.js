import { createReducer } from 'redux-create-reducer'
import { actions } from './actions'

const initialState = {
  list: []
}

const currencyReducer = createReducer(initialState, {
  [actions.REFRESH_LIST] (state, action) {
    return {
      ...state,
      list: action.data
    }
  }
})

export default currencyReducer
