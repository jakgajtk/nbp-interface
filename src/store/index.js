import { createStore, applyMiddleware } from 'redux'
import { middleware as beesMiddleware } from 'redux-bees'
import { createLogger } from 'redux-logger'

import reducer from './reducer'

const store = createStore(reducer, applyMiddleware(createLogger(), beesMiddleware()))

export default store
