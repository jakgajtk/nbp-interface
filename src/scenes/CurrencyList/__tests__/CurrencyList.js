import React from 'react'
import CurrencyList from '../../CurrencyList'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import { Provider } from 'react-redux'
import store from '../../../store'

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><MuiThemeProvider><CurrencyList /></MuiThemeProvider></Provider>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})