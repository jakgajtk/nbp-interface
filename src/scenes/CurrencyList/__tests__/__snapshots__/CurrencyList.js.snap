// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "muiPrepared": true,
        "padding": "0px 0px 8px 0px",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 14,
          "fontWeight": 500,
          "lineHeight": "48px",
          "muiPrepared": true,
          "paddingLeft": 16,
          "width": "100%",
        }
      }
    >
      <span>
        Currency Rates List
      </span>
      <span
        style={
          Object {
            "float": "right",
            "lineHeight": "30px",
          }
        }
      >
        <div
          className={undefined}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "minWidth": 88,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <button
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "#00bcd4",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": "100%",
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                style={
                  Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "margin": 0,
                      "muiPrepared": true,
                      "opacity": 1,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "userSelect": "none",
                    }
                  }
                >
                  Primary Currency Rates
                </span>
              </div>
            </div>
          </button>
        </div>
        <div
          className={undefined}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "minWidth": 88,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <button
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "#ff4081",
                "border": 10,
                "borderRadius": 2,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "height": 36,
                "lineHeight": "36px",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "relative",
                "textAlign": "center",
                "textDecoration": "none",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": "100%",
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <div
                style={
                  Object {
                    "backgroundColor": false,
                    "borderRadius": 2,
                    "height": 36,
                    "muiPrepared": true,
                    "top": 0,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "margin": 0,
                      "muiPrepared": true,
                      "opacity": 1,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "userSelect": "none",
                    }
                  }
                >
                  Secondary Currency Rates
                </span>
              </div>
            </div>
          </button>
        </div>
      </span>
    </div>
  </div>
</div>
`;
